app/policies/booking_policy.rb

```diff
@@ -11,20 +11,17 @@ def create?
   end
 
   def show?
-    record.user == user
-    # this only allows the renting person to see it. Not the owner of the dress. code below was my attempt to fix that.
+    record.dress.user == user || record.user == user
+    # the owner and renter can see it
   end
 
-  # def update?
-  #   dress = Dress.find(record.dress_id)
-  #   owner = dress.user_id
-  #   record.user == user || owner == user
-  #   # This logic should only let the owner of the dress and person who made the booking auhtorized?
-  # end
+  def update?
+    record.dress.user == user || record.user == user
+    # the owner and renter can see it
+  end
 
   def destroy?
-    record.user == user
```

Comment:
wouldn't update also allow them to change the status? I think the way it is now makes sense though, with just destroy being the strictest.

Reply:
Hi, I'm a little bit confused because your comment is on the old code.

Does "the way it is now" refer to the destroy being enabled "if" the record.user(person who made the booking) and user(current_user) is the same, and keeping the update commented out?

In the changed code, update would let owners and the renter update and destroy is false so that no "user" can destroy a booking.(I'll add the false in a second to the destroy?)

Sorry for being verbose, just trying to clarify ^_^

Reply:
Your profile looks great. Please use your page. Could we merge it and then you can take any of the logic you think works and then we could even take away the booking_index. That would make things a little easier for routing and testing.
I also, think the logic on the index page would work as one "if else" statement than two "if" statements if you want to use it ^_^